docs/architecture/adr-001-umee-module.md

```diff
@@ -0,0 +1,90 @@
+# ADR 001: Umee Module
+
+## Changelog
+
+- April 28, 2021: Initial Draft (@alexanderbez)
+
+## Status
+
+Proposed
+
+## Context
+
+Umee is a Universal Capital Facility that can collateralize assets on one
+blockchain towards borrowing assets on another blockchain. The platform specializes
+in allowing staked assets from POS blockchains to be used as collateral for
+borrowing across blockchains. The platform uses a combination of algorithmically
+determined interest rates based on market driven conditions.
+
+For the initial MVP implementation of the Umee network, we require the ability
+for users to be able to send ATOM tokens to a dedicated pool in the Umee network
+via IBC. ATOM tokens deposited into the Umee pool will mint a derivative meToken
+in a one-to-one ratio, i.e. one ATOM mints one meToken derivative.
+
+Validators on the Umee network will take these deposited ATOM tokens and delegate
+them to a set of governance-controlled validators on the ATOM source chain,
+e.g. the Cosmos Hub.
+
+A user that sent ATOM tokens to the Umee network can then take their derivative
+meTokens and send them to Ethereum via a bridge where a synthetic ERC-20 version
+of the meTokens are minted. Once sent, the meTokens are locked in Umee and the
+user can then freely trade and operate within Ethereum's DeFi ecosystem with the
+synthetic ERC-20 meTokens.
+
+## Decision
+
+To support users sending ATOM tokens to the Umee network via IBC, we will create
+a `x/umee` module that contains a dedicated `ModuleAccount` that represents the
+ATOM pool. In addition, we will implement a [custom IBC module](https://github.com/cosmos/ibc-go/blob/v1.0.0-alpha1/docs/custom.md).
+The custom IBC module will handle receiving custom packets on a custom Umee port.
+The custom module will be nearly identical to the ICS-20 module, except that it
+will perform the following:
+
+- Automatically send source ATOM tokens to the dedicated `ModuleAccount` pool
+  address instead of a user supplied address.
+- Mark the transfer's amount in a pending queue, `PendingDelegationQueue`, to be
+  later used for delegation on the source chain.
```

Comment:
do we only support the cosmos hub? what happens if an external chain, say Iris, also connects (independently) to Umee and performs a transfer? I guess the chains supported by Umee will have to be registered in the parameters. Also we need to take into consideration the naming collision for the chain ID in this case

Reply:
I guess we would also need to have a pending delegation queue by base denom type (ATOM, etc)

Reply:
We will only support the Hub (for now). Multi-chain becomes much more complex.

Reply:
Makes sense, the only thing is that you somehow have to prevent other chains from doing the same (i.e only allow IBC connections with the umee module from the Hub)